### DISCUSSION TITLE
Meaning of phase in Meep

### DISCUSSION THREAD

**Discussion by harnessa** [NanoComp/meep#1913] [Q&A] [answered]
I am curious about how well I can trust the relative phase of a field in Meep. 

Here is an outline of my simulation: I am working in 2D and am propagating a plane wave through the air gap between two sides of a metal coated silicon wafer (Gaussian source wavelength: 0.6 microns, gap width >10 microns, Al coating: 0.25 microns thick, cSi wafer: 2 microns thick). I want to calculate the amplitude and phase change at the bottom of the wafer (downstream of the source) that is caused by the presence of the wafer. I run one simulation with the wafer and save the DFT_fields from the bottom of the wafer (I'll call these wafer_fields). I run a second simulation with all materials removed (vacuum simulation) and save the DFT_fields at the same location at the bottom of the wafer (I'll call these vacuum_fields). Both simulations are run for the same amount of time (occasionally they differ by 1 timestep). 

The amplitude and phase change is calculated as: wafer_fields / vacuum_fields - 1. Dividing by vacuum_fields normalizes the amplitude and subtracts the phase that would occur due to propagation only. 

My question is: can that relative phase change be trusted? Is this the correct way to get the phase change? This approach works when compared to the analytic solution of a plane wave propagating through a gap in a PEC screen, but will this approach still hold when using a cSi wafer? Am I missing something or being overly skeptic?

Thanks,

**Comment by stevengj** [comment]
It sounds fine.  (I'm not entirely clear on your geometry, however.  Note that the "input and output" media should probably be the same between your first and second simulations — in your wafer simulation, is the light entering and exiting from vacuum, or from a semi-infinite Si medium?)

Also, note that you don't need (or necessarily want) to run for the same amount of time in the two simulations — you just need to run long enough for the DFT (the Fourier transform) to converge, i.e. until the time-domain fields have decayed away.  (Depending on the structure, you might need to run for longer, e.g. if there is a resonance that takes a long time to decay.)

**Reply by harnessa** [reply]
Below is a figure of the geometry. The red dotted line is the location of the source (and propagating with positive X, i.e., 'down') . The black dotted line is where I am most interested in the field. Light enters and exits from vacuum. 

![eps](https://user-images.githubusercontent.com/39776172/150557316-da541760-16f5-49ee-a9dd-b90881af94d0.png)

Yes, I run until the time-domain fields have decayed away in the wafer simulation. Originally I did the same for the vacuum simulation, but was concerned about a time difference in the phase, so I ran for the same amount of time as the wafer simulation. There was not a meaningful difference between the too.

Thanks

**Comment by oskooi** [comment]
For reference, another way to set up this calculation is demonstrated in [Tutorial/Focusing Properties of a Metasurface Lens](https://meep.readthedocs.io/en/latest/Python_Tutorials/Near_to_Far_Field_Spectra/#focusing-properties-of-a-metasurface-lens). This example involves computing the transmittance and relative phase (using mode decomposition) of the zeroth diffraction order of a unit cell of a metalens as a function of its duty cycle. Note that the [`get_eigenmode_coefficients`](https://meep.readthedocs.io/en/latest/Python_User_Interface/#mode-decomposition) function can be passed a [`DiffractedPlanewave`](https://meep.readthedocs.io/en/latest/Python_User_Interface/#diffractedplanewave) object (rather than the band number which mainly applies to guided modes) for specifying the diffraction order in 3d.

**Reply by harnessa** [reply]
Could that approach still be used if the geometry was a one-sided edge? (i.e., the wafer on the right side is removed)

**Comment by oskooi** [comment, marked as answer]
Yes. More generally, in the context of coupling from one mode to another, you can take the ratio of the two complex mode coefficients which is defined as the [scattering (S) parameter](https://en.wikipedia.org/wiki/Scattering_parameters). The phase of the S-parameter (itself a complex quantity) is equivalent to the relative phase of the two modes at the two monitors.
